### DISCUSSION TITLE
How to bundle multiple JSON schemas into a single typescript file?

### DISCUSSION THREAD

**Discussion by x12a1f** [quicktype/quicktype#2458] [Q&A] [answered]
I have a couple of JSON schemas for which I want to generate typescript for. This works but I end up with a typescript file for each JSON schema, each with its own Convert class. I would like to bundle all the JSON schemas into a single typescript file but I'm unsure how to do this.

The "Multiple JSON" option works for plain JSON files and generates a single typescript file for a bunch of JSON files. I'm looking for something similar but for JSON schemas.

**Comment by kriswest** [comment, marked as answer]
I believe there is a bug in command line argument handling for the source type (-s) flag when passing a directory of files so we pass each schema file individually and use a command line script to make that easier (reads the dir and constructs the command with the files it finds): https://github.com/finos/FDC3/blob/master/s2tQuicktypeUtil.js

Perhaps all you needed to know is that you can append multiple source schema files to the end of your command line call, with a specified output file, and they will all output into it.

**Reply by x12a1f** [reply]
That indeed does what I need it to do... I'm not sure why I didn't think of trying it..
